### ISSUE TITLE
[feat] flag nested render helper functions inside component bodies

### ISSUE THREAD

**Issue by SukkaW** [Rel1cx/eslint-react#1607]
### Problem Description

```tsx
function Comp() {
  const config = useXXX();

  const renderEl = (config) => <span>{config}</span>;

  return <div>{renderEl(config)}</div>;
}
```

This pattern is bad. This is just a nested component or an IIFE in disguise.

Currently, `no-nested-component-definitions` can not flag this.

### Alternative Solutions

N/A

### Feature Details

Maybe we can update `no-nested-component-definitions` to match "render function" as well.

### Examples

See "Problem Description" section.

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users
